services/worker/src/worker/job_runners/parquet_and_dataset_info.py

```diff
@@ -423,6 +447,202 @@ class EmptyFeaturesError(Exception):
     pass
 
 
+class DatasetWithTooManyExternalFilesError(ParquetAndDatasetInfoJobRunnerError):
+    """Raised when the dataset size (sum of config sizes given by the datasets library) is too big."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.NOT_IMPLEMENTED, "DatasetWithTooManyExternalFilesError", cause, False)
```

Comment:
For all these new errors, we should disclose the exception since the dataset maintainer can fix the issue. Same for all the errors below.

```suggestion
        super().__init__(message, HTTPStatus.NOT_IMPLEMENTED, "DatasetWithTooManyExternalFilesError", cause, True)
```